cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -105,16 +105,17 @@ func (c *CStorPoolController) syncHandler(key string, operation common.QueueOper
 				err.Error(),
 			),
 		)
-		cspObject.VersionDetails.Status.Message = "Failed to reconcile csp version"
-		cspObject.VersionDetails.Status.Reason = err.Error()
-		cspObject.VersionDetails.Status.LastUpdateTime = metav1.Now()
+		cspObject.VersionDetails.Status = upgrade.SetErrorStatus(
```

Comment:
why to assign back? can't it be call by reference method?

Reply:
This will require writing the functions in `pkg/apis/openebs.io/v1alpha1` folder as we can not write methods on non-local structs. Is that ok ?